src/actionsHandler.ts

```diff
@@ -0,0 +1,232 @@
+import { PrismaClient } from '@prisma/client';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { extractHashOrdinal, extractPagination } from './request-params';
+import { paginatedQuery, fromCreatedAtOrdinalCursor, toCreatedAtOrdinalCursor } from './pagination';
+import { respond, handleError, dagTransactionResponse } from './response';
+
+const prisma = new PrismaClient();
+
+
+// const prisma = new PrismaClient({
+//   log: [
+//     {
+//       emit: "event",
+//       level: "query",
+//     },
+//   ],
+// });
+
+// prisma.$on("query", async (e) => {
+//     console.log(`${e.query} ${e.params}`)
+// });
+
+const dagActionsTables =  [
+    "dag_allow_spends",
+    "dag_spend_transactions",
+    "dag_token_locks",
+    "dag_token_unlocks",
+    "dag_fee_transactions"
+  ];
+
+  const metagraphActionsTables = [
+    "metagraph_allow_spends",
+    "metagraph_spend_transactions",
+    "metagraph_token_locks",
+    "metagraph_token_unlocks",
+    "metagraph_fee_transactions"
+  ];
+
+  const currencyId = (transaction) => (transaction.metagraph_token_lock?.metagraph_id ?? 
+    transaction.metagraph_token_unlock?.metagraph_id ?? 
+    transaction.metagraph_allow_spend?.metagraph_id ?? 
+    transaction.metagraph_spend_transaction?.metagraph_id ??
+    transaction.metagraph_fee_transaction?.metagraph_id)
+  
+  const actionResponse = (transaction) => ({
+    type: transaction.table_name,
```

Comment:
Let's strip the "s" off the end of the table name and remove the "dag_" or "metagraph_" prefixes. 

So "dag_allow_spends" becomes "allow_spend", etc.